modules/currency-l1/src/main/scala/org/tessellation/currency/l1/domain/dataApplication/consensus/Engine.scala

```diff
@@ -323,16 +326,25 @@ object Engine {
               }
             }
 
-          roundData
-            .formBlock(
-              a => dataApplication.validateUpdate(a).map(_.toEither.leftMap(_.toString).map(_ => a)),
-              combine
-            )
-            .flatMap {
+          def validateFn: F[Signed[DataUpdate] => F[DataApplicationValidationErrorOr[Unit]]] =
+            OptionT(lastGlobalSnapshot.getOrdinal).map { gsOrdinal => (update: Signed[DataUpdate]) =>
```

Comment:
@marcinwadon it does. The function returns `Signed[DataUpdate] => F[DataApplicationValidationErrorOr[Unit]]` in an `F`

Reply:
yeah, so it can be a function from `Signed[DataUpdate]` to `F[DataApplicationValidationErrorOr[Unit]]` so you don't need to wrap it into an `F` and just use it in flatMap or sth 🤔 it looks like an arbitrary lift operation that is not needed, but I might not know the full context

Reply:
@marcinwadon The function produces the function as you said but it requires other arguments. Right now it needs the global snapshot ordinal which has a return type of `F[Option[SnapshotOrdinal]]`. I can keep that here or I need to move that code before I create the validate function. Whichever you prefer.

Reply:
@marcinwadon I inlined it.